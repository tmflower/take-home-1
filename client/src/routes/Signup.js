import { useState } from "react";
import axios from "axios";

export function Signup({ setUserAdded }) {
    const [message, setMessage] = useState('')
    const [formData, setFormData] = useState({
        firstName: '',
        lastName: '',
        email: ''
    });

    const { firstName, lastName, email } = formData;

    const handleChange = (evt) => {
        const { name, value } = evt.target;
        setFormData(formData => ({...formData, [name]: value}));
    };

    const handleSubmit = async(evt) => {
        evt.preventDefault();
        try { 
            await axios.post('http://localhost:3001/users', { firstName, lastName, email });
            setMessage(`Welcome to Yodlr, ${firstName}!`);
            setUserAdded(true);
        }
        catch(err) {
            console.error(err);
        }
    }
    return (
        <div>
            <h1>Yodlr Registration Portal</h1>
            {message ? <p>{message}</p> : null}
            <form onSubmit={handleSubmit}>
                <label htmlFor="firstName">
                    First name:
                    <input
                        type="text"
                        name="firstName"
                        value={firstName}
                        id="firstName"
                        onChange={handleChange}>
                    </input>                    
                </label>
                <label htmlFor="lastName">
                    Last name:
                    <input
                        type="text"
                        name="lastName"
                        value={lastName}
                        id="lastName"
                        onChange={handleChange}>
                    </input>
                </label>
                <label htmlFor="email">
                    Email:
                    <input
                        type="email"
                        name="email"
                        value={email}
                        id="email"
                        onChange={handleChange}>
                    </input>
                </label>
            <button type="submit">Submit</button>
            </form>
        </div>
    )
}