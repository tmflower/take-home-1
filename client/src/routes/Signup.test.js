import { render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Signup } from './Signup';
import userEvent from "@testing-library/user-event";
import axios from 'axios';

jest.mock('axios');

test('renders Home without crashing', () => {
    render(
        <MemoryRouter>
            <Signup/>
        </MemoryRouter>);
});

test('matches snapshot', () => {
    const { asFragment } =
    render(
        <MemoryRouter>
            <Signup />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

test('renders Signup correctly', () => {
    render(
        <MemoryRouter>
            <Signup />
        </MemoryRouter>
    );
    const h1 = screen.getByText('Yodlr Registration Portal');
    expect(h1).toBeInTheDocument();
});

test('handles submit button click', async () => {
    render(
        <MemoryRouter>
            <Signup setUserAdded={() => console.log("user added")}/>
        </MemoryRouter>
    );

    const messageBefore = screen.queryByText(`Welcome to Yodlr, !`);
    expect(messageBefore).not.toBeInTheDocument();

    userEvent.click(screen.getByRole('button', { name: 'Submit' }));

    await waitFor(() => {
        const messageAfter = screen.queryByText(`Welcome to Yodlr, !`);
        expect(messageAfter).toBeInTheDocument();
    });
});