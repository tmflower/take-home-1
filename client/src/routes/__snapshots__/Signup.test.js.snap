// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-dvxtzn"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2 css-1b3u51d-MuiTypography-root"
    >
      Yodlr Registration Portal
    </h2>
    <form>
      <div
        class="MuiFormControl-root css-16hka9-MuiFormControl-root"
      >
        <div
          class="css-1p5q5e5-MuiStack-root"
        >
          <label
            for="firstName"
          >
            First name:
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="firstName"
                  name="firstName"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </label>
          <label
            for="lastName"
          >
            Last name:
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="lastName"
                  name="lastName"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </label>
          <label
            for="email"
          >
            Email:
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="email"
                  name="email"
                  type="email"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </label>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-usvop9-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            Submit
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
